esrally/driver/runner.py

```diff
@@ -793,111 +771,147 @@ async def request_body_query(self, es, params):
         # disable eager response parsing - responses might be huge thus skewing results
         es.return_raw_response()
 
-        r = await self._raw_search(es, doc_type, index, body, request_params, headers=headers)
+        async def search_after_query(es, params):
+            index = params.get("index", "_all")
+            pit_op = params.get("with-point-in-time-from")
+            results = {
+                "unit": "pages",
+                "success": True,
+                "took": 0
+            }
+            if pit_op:
+                # these are disallowed as they are encoded in the pit_id
+                for item in ["index", "routing", "preference"]:
+                    body.pop(item, None)
+                index = None
+            # explicitly convert to int to provoke an error otherwise
+            total_pages = sys.maxsize if params.get("pages") == "all" else int(params.get("pages"))
+            for page in range(1, total_pages + 1):
+                if pit_op:
+                    pit_id = CompositeContext.get(pit_op)
+                    body["pit"] = {"id": pit_id,
+                                   "keep_alive": "1m" } # todo parameterize second keep_alive
+
+                response = await self._raw_search(es, doc_type=None, index=index, body=body.copy(), params=request_params, headers=headers)
+                parsed, last_sort = parsing.extract_search_after_properties(response, bool(pit_op), results.get("hits"))
+                results["pages"] = page
+                results["weight"] = page
+                if results.get("hits") is None:
+                    results["hits"] = parsed.get("hits.total.value")
+                    results["hits_relation"] = parsed.get("hits.total.relation")
+                results["took"] += parsed.get("took")
+                results["timed_out"] = parsed.get("timed_out")
+                if pit_op:
+                    # per the documentation the response pit id is most up-to-date
+                    CompositeContext.put(pit_op, parsed.get("pit_id"))
+
+                if results.get("hits") / size > page:
+                    body["search_after"] = last_sort
+                else:
+                    # done. clean body for next iteration
+                    for item in ["pit", "search_after"]:
+                        body.pop(item, None)
```

Comment:
took this route to avoid a deep copy of the body parameter.   personally i think the `query` should be the operation parameter, and the body assembled within the runner, then we wouldn't have this issue.  open to other ideas